integration/backfill_test.go

```diff
@@ -184,31 +172,28 @@ func verifyBlock(t *testing.T, client objstore.Bucket, ulid ulid.ULID, localPath
 	require.Equal(t, localMeta.Stats, remoteMeta.Stats)
 }
 
-func runMimirtoolBackfill(t *testing.T, args ...string) (string, error) {
-	oldStdout, oldStderr := os.Stdout, os.Stderr
-	defer func() {
-		os.Stdout = oldStdout
-		os.Stderr = oldStderr
-
-		logrus.SetOutput(os.Stderr)
-	}()
-
-	co := util_test.CaptureOutput(t)
-	// mimirtool uses logrus, which has global logger. We reinitialize it to use our captured Stderr.
-	logrus.SetOutput(os.Stderr)
-
-	app := kingpin.New("mimirtool", "help")
-	envVars := commands.NewEnvVarsWithPrefix("MIMIR")
+func runMimirtoolBackfill(sharedDir string, compactor *e2emimir.MimirService, blocksToUpload ...ulid.ULID) (string, error) {
+	dockerArgs := []string{
+		"run",
+		"--rm",
+		"--net=" + networkName,
+		"--name=" + networkName + "-mimirtool-backfill",
+	}
+	dockerArgs = append(dockerArgs, "--volume", fmt.Sprintf("%s:%s:z", sharedDir, e2e.ContainerSharedDir))
+	dockerArgs = append(dockerArgs, e2emimir.GetMimirtoolImage())
 
-	// Only register mimirtool backfill command.
-	backfill := commands.BackfillCommand{}
-	backfill.Register(app, envVars)
+	// Mimirtool args.
+	dockerArgs = append(dockerArgs, "backfill")
+	dockerArgs = append(dockerArgs, "--address", fmt.Sprintf("http://%s:%d/", compactor.Name(), compactor.HTTPPort()))
+	dockerArgs = append(dockerArgs, "--id", "anonymous")
 
-	// Run mimirtool (Parse calls any action).
-	_, err := app.Parse(append([]string{"backfill"}, args...))
+	for _, b := range blocksToUpload {
+		dockerArgs = append(dockerArgs, path.Join(e2e.ContainerSharedDir, b.String()))
+	}
 
-	stdout, stderr := co.Done()
-	return stdout + stderr, err
+	out, err := e2e.RunCommandAndGetOutput("docker", dockerArgs...)
+	fmt.Println(string(out))
```

Comment:
Is it intentional to keep it? I'm fine with that, just double checking it's not a leftover.

Reply:
It's a leftover actually. Let's remove, we can always put it back when debugging output.